<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Untitled â€¢ cqsR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Untitled">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">cqsR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/AdvancedR.html">Advanced R</a>
    </li>
    <li>
      <a href="../articles/EasierStatisticModels.html">Statistic Methods Example</a>
    </li>
    <li>
      <a href="../articles/limmaTestAndDesign.html">limma Test and Design</a>
    </li>
    <li>
      <a href="../articles/matchVcfToGr.html">Untitled</a>
    </li>
    <li>
      <a href="../articles/RandSQLinGoogleCloud.html">R and SQL in google cloud</a>
    </li>
    <li>
      <a href="../articles/reportExample.html">Report Example</a>
    </li>
    <li>
      <a href="../articles/RforDataScienceNotes.html">R for Data Science Notes</a>
    </li>
    <li>
      <a href="../articles/RnaSeqSampleSize2.html">Development of RnaSeqSampleSize version 2</a>
    </li>
    <li>
      <a href="../articles/RPackageExamples.html">R Packages Examples</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Untitled</h1>
            
      
      
      <div class="hidden name"><code>matchVcfToGr.Rmd</code></div>

    </div>

    
    
<div id="matchgrtovcfobj" class="section level1">
<h1 class="hasAnchor">
<a href="#matchgrtovcfobj" class="anchor"></a>matchGrToVcfObj</h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" title="1"><span class="co">#library(VariantAnnotation)</span></a>
<a class="sourceLine" id="cb1-2" title="2"><span class="co">#library(data.table)</span></a>
<a class="sourceLine" id="cb1-3" title="3"><span class="co">#devtools::load_all("d:/source/cqsR")</span></a>
<a class="sourceLine" id="cb1-4" title="4"><span class="co">#devtools::load_all(".")</span></a>
<a class="sourceLine" id="cb1-5" title="5"></a>
<a class="sourceLine" id="cb1-6" title="6"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(cqsR)</a></code></pre></div>
<pre><code>## Loading required package: data.table</code></pre>
<pre><code>## Warning: package 'data.table' was built under R version 3.6.1</code></pre>
<pre><code>## Loading required package: ggplot2</code></pre>
<pre><code>## Warning: package 'ggplot2' was built under R version 3.6.3</code></pre>
<pre><code>## Loading required package: htmlTable</code></pre>
<pre><code>## Warning: package 'htmlTable' was built under R version 3.6.1</code></pre>
<pre><code>## Loading required package: Hmisc</code></pre>
<pre><code>## Warning: package 'Hmisc' was built under R version 3.6.3</code></pre>
<pre><code>## Loading required package: lattice</code></pre>
<pre><code>## Loading required package: survival</code></pre>
<pre><code>## Warning: package 'survival' was built under R version 3.6.3</code></pre>
<pre><code>## Loading required package: Formula</code></pre>
<pre><code>## 
## Attaching package: 'Hmisc'</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     format.pval, units</code></pre>
<pre><code>## Loading required package: VariantAnnotation</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## Loading required package: parallel</code></pre>
<pre><code>## 
## Attaching package: 'BiocGenerics'</code></pre>
<pre><code>## The following objects are masked from 'package:parallel':
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     anyDuplicated, append, as.data.frame, basename, cbind,
##     colnames, dirname, do.call, duplicated, eval, evalq, Filter,
##     Find, get, grep, grepl, intersect, is.unsorted, lapply, Map,
##     mapply, match, mget, order, paste, pmax, pmax.int, pmin,
##     pmin.int, Position, rank, rbind, Reduce, rownames, sapply,
##     setdiff, sort, table, tapply, union, unique, unsplit, which,
##     which.max, which.min</code></pre>
<pre><code>## Loading required package: GenomeInfoDb</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## 
## Attaching package: 'S4Vectors'</code></pre>
<pre><code>## The following objects are masked from 'package:data.table':
## 
##     first, second</code></pre>
<pre><code>## The following object is masked from 'package:base':
## 
##     expand.grid</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## 
## Attaching package: 'IRanges'</code></pre>
<pre><code>## The following object is masked from 'package:data.table':
## 
##     shift</code></pre>
<pre><code>## The following object is masked from 'package:grDevices':
## 
##     windows</code></pre>
<pre><code>## Loading required package: GenomicRanges</code></pre>
<pre><code>## Loading required package: SummarizedExperiment</code></pre>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     'browseVignettes()'. To cite Bioconductor, see
##     'citation("Biobase")', and for packages 'citation("pkgname")'.</code></pre>
<pre><code>## 
## Attaching package: 'Biobase'</code></pre>
<pre><code>## The following object is masked from 'package:Hmisc':
## 
##     contents</code></pre>
<pre><code>## Loading required package: DelayedArray</code></pre>
<pre><code>## Loading required package: matrixStats</code></pre>
<pre><code>## Warning: package 'matrixStats' was built under R version 3.6.1</code></pre>
<pre><code>## 
## Attaching package: 'matrixStats'</code></pre>
<pre><code>## The following objects are masked from 'package:Biobase':
## 
##     anyMissing, rowMedians</code></pre>
<pre><code>## Loading required package: BiocParallel</code></pre>
<pre><code>## 
## Attaching package: 'DelayedArray'</code></pre>
<pre><code>## The following objects are masked from 'package:matrixStats':
## 
##     colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     aperm, apply, rowsum</code></pre>
<pre><code>## Loading required package: Rsamtools</code></pre>
<pre><code>## Loading required package: Biostrings</code></pre>
<pre><code>## Loading required package: XVector</code></pre>
<pre><code>## 
## Attaching package: 'Biostrings'</code></pre>
<pre><code>## The following object is masked from 'package:DelayedArray':
## 
##     type</code></pre>
<pre><code>## The following objects are masked from 'package:Hmisc':
## 
##     mask, translate</code></pre>
<pre><code>## The following object is masked from 'package:base':
## 
##     strsplit</code></pre>
<pre><code>## 
## Attaching package: 'VariantAnnotation'</code></pre>
<pre><code>## The following object is masked from 'package:base':
## 
##     tabulate</code></pre>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb57-1" title="1">fl &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"chr22.vcf.gz"</span>, <span class="dt">package=</span><span class="st">"VariantAnnotation"</span>)</a>
<a class="sourceLine" id="cb57-2" title="2">vcf &lt;-<span class="st"> </span><span class="kw">readVcf</span>(fl, <span class="st">"hg19"</span>)</a>
<a class="sourceLine" id="cb57-3" title="3">vcf &lt;-<span class="st"> </span>VariantAnnotation<span class="op">::</span><span class="kw">expand</span>(<span class="dt">x =</span> vcf, <span class="dt">row.names =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb57-4" title="4"></a>
<a class="sourceLine" id="cb57-5" title="5">variantsTable=<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/data.frame">data.frame</a></span>(<span class="dt">chr=</span><span class="st">"22"</span>, <span class="dt">start=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">50300078</span>,<span class="dv">50300085</span>), <span class="dt">end=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">50300078</span>,<span class="dv">50300102</span>), <span class="dt">score=</span><span class="dv">1</span><span class="op">:</span><span class="dv">2</span>)</a>
<a class="sourceLine" id="cb57-6" title="6">variantsTableGr=<span class="kw">makeGRangesFromDataFrame</span>(variantsTable)</a>
<a class="sourceLine" id="cb57-7" title="7"><span class="co">#matchGrToVcfObj(variantsTableGr,vcf)</span></a>
<a class="sourceLine" id="cb57-8" title="8"><span class="kw">readVcfAndMatchGr</span>(fl,variantsTableGr)</a></code></pre></div>
<pre><code>## [1] "Reading vcf: 10376"</code></pre>
<pre><code>##    chr    start      end REF ALT FILTER   AF
## 1:  22 50300078 50300078   A   G   PASS 0.34
## 2:  22 50300086 50300086   C   T   PASS 0.01</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#matchgrtovcfobj">matchGrToVcfObj</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by The package maintainer.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
